diff --git a/env_support/cmake/custom.cmake b/env_support/cmake/custom.cmake
index 561698211..77d9296c0 100644
--- a/env_support/cmake/custom.cmake
+++ b/env_support/cmake/custom.cmake
@@ -63,6 +63,12 @@ install(
   FILES_MATCHING
   PATTERN "*.h")
 
+install(
+  DIRECTORY "${CMAKE_SOURCE_DIR}/demos"
+  DESTINATION "${CMAKE_INSTALL_PREFIX}/${INC_INSTALL_DIR}/src/"
+  FILES_MATCHING
+  PATTERN "*.h" "*.c")
+
 set_target_properties(
   lvgl
   PROPERTIES OUTPUT_NAME lvgl
